pulp_ostree/app/models.py

```diff
@@ -122,14 +134,39 @@ class OstreeRepository(Repository):
 
     TYPE = "ostree"
 
-    CONTENT_TYPES = [OstreeCommit, OstreeRefsHead, OstreeObject, OstreeConfig]
+    CONTENT_TYPES = [OstreeCommit, OstreeRefsHead, OstreeObject, OstreeConfig, OstreeSummary]
 
     class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
 
     def finalize_new_version(self, new_version):
-        """Handle RefsHead duplicates."""
+        """Handle repository duplicates, especially configs and summaries."""
         remove_duplicates(new_version)
+
+        if new_version.base_version:
+            previous_version = new_version.base_version
+        else:
+            if new_version.previous().number != 0:
+                previous_version = new_version.previous()
+            else:
+                previous_version = None
+
+        if previous_version:
+            new_config = new_version.added().get(pulp_type=OstreeConfig.get_pulp_type())
+            new_summary = new_version.added().get(pulp_type=OstreeSummary.get_pulp_type())
+
+            existing_files_sha256 = (
+                new_config._artifacts.get().sha256, new_summary._artifacts.get().sha256
+            )
+
+            existing_content = previous_version.added().filter(
```

Comment:
shouldn*t that be present instead of added?

Reply:
No, because I am filtering the old content based on sha256. When something is found, I am then using it for determining which content units are existing and need to be removed (i.e., `content_to_remove = new_repo_version.added(pk__in=(new_summary.pk, new_config.pk) and not in existing_content)`; this means that only content units with a new pk will be removed, or not?).

Reply:
No, i mean if the thing that is there was already added in the second to last version, this query would miss it.